File: tl/src/main/java/com/github/badoualy/telegram/tl/api/requests/TLRequestMessagesSaveGif.java

package com.github.badoualy.telegram.tl.api.requests;

import com.github.badoualy.telegram.tl.TLContext;
import com.github.badoualy.telegram.tl.core.TLBool;
import com.github.badoualy.telegram.tl.core.TLMethod;
import com.github.badoualy.telegram.tl.core.TLObject;

import java.io.IOException;
import java.io.InputStream;
import java.io.OutputStream;

import static com.github.badoualy.telegram.tl.StreamUtils.readTLBool;
import static com.github.badoualy.telegram.tl.StreamUtils.readTLObject;
import static com.github.badoualy.telegram.tl.StreamUtils.writeTLBool;
import static com.github.badoualy.telegram.tl.StreamUtils.writeTLObject;


public class TLRequestMessagesSaveGif extends TLMethod<TLBool> {

    public static final int CLASS_ID = 0x327a30cb;

    public int getClassId() {
        return CLASS_ID;
    }


    public TLRequestMessagesSaveGif(        com.github.badoualy.telegram.tl.api.TLAbsInputDocument _id,         boolean _unsave) {
        this.id = _id;
        this.unsave = _unsave;

    }



    public TLBool deserializeResponse(InputStream stream, TLContext context) throws IOException {

        TLObject res = readTLObject(stream, context);
        if (res == null) {
            throw new IOException("Unable to parse response");
        }
        if (res instanceof TLBool) {
            return (TLBool)res;
        }
        else {
            throw new IOException("Incorrect response type. Expected TLBool, got: " + res.getClass().getCanonicalName());
        }

    }
        


    protected com.github.badoualy.telegram.tl.api.TLAbsInputDocument id;

    protected boolean unsave;


    public com.github.badoualy.telegram.tl.api.TLAbsInputDocument getId() {
        return id;
    }

    public void setId(com.github.badoualy.telegram.tl.api.TLAbsInputDocument value) {
        this.id = value;
    }

    public boolean getUnsave() {
        return unsave;
    }

    public void setUnsave(boolean value) {
        this.unsave = value;
    }


    @Override
    public void serializeBody(OutputStream stream) throws IOException {

        writeTLObject(this.id, stream);
        writeTLBool(this.unsave, stream);
    }


    @Override
    public void deserializeBody(InputStream stream, TLContext context) throws IOException {

        this.id = (com.github.badoualy.telegram.tl.api.TLAbsInputDocument)readTLObject(stream, context);
        this.unsave = readTLBool(stream);
    }



    @Override
    public String toString() {
        return "messages.saveGif#327a30cb";
    }

}

File: tl/src/main/kotlin/com/github/badoualy/telegram/tl/api/request/TLRequestMessagesSaveGif.kt
package com.github.badoualy.telegram.tl.api.request

import com.github.badoualy.telegram.tl.TLObjectUtils.SIZE_BOOLEAN
import com.github.badoualy.telegram.tl.TLObjectUtils.SIZE_CONSTRUCTOR_ID
import com.github.badoualy.telegram.tl.TLObjectUtils.SIZE_DOUBLE
import com.github.badoualy.telegram.tl.TLObjectUtils.SIZE_INT32
import com.github.badoualy.telegram.tl.TLObjectUtils.SIZE_INT64
import com.github.badoualy.telegram.tl.TLObjectUtils.computeTLBytesSerializedSize
import com.github.badoualy.telegram.tl.TLObjectUtils.computeTLStringSerializedSize
import com.github.badoualy.telegram.tl.api.TLAbsInputDocument
import com.github.badoualy.telegram.tl.api.TLInputDocumentEmpty
import com.github.badoualy.telegram.tl.core.TLBool
import com.github.badoualy.telegram.tl.core.TLMethod
import com.github.badoualy.telegram.tl.serialization.TLDeserializer
import com.github.badoualy.telegram.tl.serialization.TLSerializer
import java.io.IOException
import kotlin.Any
import kotlin.Boolean
import kotlin.Int
import kotlin.String
import kotlin.jvm.Throws

/**
 * @author Yannick Badoual yann.badoual@gmail.com
 * @see <a href="http://github.com/badoualy/kotlogram">http://github.com/badoualy/kotlogram</a>
 */
class TLRequestMessagesSaveGif() : TLMethod<TLBool>() {
    var id: TLAbsInputDocument = TLInputDocumentEmpty()

    var unsave: Boolean = false

    private val _constructor: String = "messages.saveGif#327a30cb"

    override val constructorId: Int = CONSTRUCTOR_ID

    constructor(id: TLAbsInputDocument, unsave: Boolean) : this() {
        this.id = id
        this.unsave = unsave
    }

    @Throws(IOException::class)
    override fun serializeBody(tlSerializer: TLSerializer) = with (tlSerializer)  {
        writeTLObject(id)
        writeBoolean(unsave)
    }

    @Throws(IOException::class)
    override fun deserializeBody(tlDeserializer: TLDeserializer) = with (tlDeserializer)  {
        id = readTLObject<TLAbsInputDocument>()
        unsave = readBoolean()
    }

    override fun computeSerializedSize(): Int {
        var size = SIZE_CONSTRUCTOR_ID
        size += id.computeSerializedSize()
        size += SIZE_BOOLEAN
        return size
    }

    override fun toString() = _constructor

    override fun equals(other: Any?): Boolean {
        if (other !is TLRequestMessagesSaveGif) return false
        if (other === this) return true

        return id == other.id
                && unsave == other.unsave
    }
    companion object  {
        const val CONSTRUCTOR_ID: Int = 0x327a30cb.toInt()
    }
}
